src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -209,30 +209,46 @@ internal static byte[] GetEncodedTruncatedBytesFromString(string? text, Encoding
             {
                 return encoding.GetBytes(text);
             }
+            
+            int textByteCount = encoding.GetByteCount(text);
+            byte[] sharedBuffer = ArrayPool<byte>.Shared.Rent(textByteCount);
+            try
+            {
+                encoding.GetBytes(text, 0, text.Length, sharedBuffer, 0);
+                var result = GetBytesFromBuffer(text, isUTF8, textByteCount, maxBytes, sharedBuffer);
+                return result;
+            }
+            finally
```

Comment:
Typically we do not recommend putting array rental and return in a try / finally. Other reviews have said as much:

https://github.com/dotnet/runtime/pull/80331#discussion_r1105041036

If an exception occurs, it's okay to not return it - the garbage collector will handle it.

Reply:
Fixed. However, some inconsistencies still leave external developers confused. 

> the garbage collector will handle it

The doc of [ArrayPool<T>.Rent()](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.rent) says,

> Failure to return a rented buffer is not a fatal error. However, it may lead to decreased application performance, as the pool may need to create a new buffer to replace the lost one.

And the example from @huoyaoyuan does use a `try-finally` pattern.

I don't mean disagreement or complain. We'd better have some kind of guidance/principles which help external developers get rid of these time-wasting traps.